Realm/RLMSyncSession.mm

```diff
@@ -52,20 +52,9 @@ - (void)suppressNextNotification {
 }
 
 - (void)invalidate {
-    if (auto session = _session.lock()) {
-        session->unregister_progress_notifier(_token);
-        _session.reset();
-        _token = 0;
-    }
-}
-
-- (void)dealloc {
-    if (_token != 0) {
-        NSLog(@"RLMProgressNotificationToken released without unregistering a notification. "
-              @"You must hold on to the RLMProgressNotificationToken and call "
-              @"-[RLMProgressNotificationToken invalidate] when you no longer wish to receive "
-              @"progress update notifications.");
-    }
+    _session->unregister_progress_notifier(_token);
```

Comment:
This needs to check if `_session` is non-null so that calling `invalidate` twice isn't a use-after-free.